app/src/main/java/com/moyerun/moyeorun_android/network/api/ApiService.kt

```diff
@@ -0,0 +1,12 @@
+package com.moyerun.moyeorun_android.network.api
+
+import com.moyerun.moyeorun_android.network.callAdapter.ApiResult
+import retrofit2.http.GET
+
+interface ApiService {
+    @GET("success")
+    suspend fun getSuccess(): ApiResult<Success<Any>>
+
+    @GET("failure")
+    suspend fun getFailure(): ApiResult<Success<Any>>
+}
```

Comment:
서버 팀과 협의해서 Success, Error 데이터 클래스를 만들어두었는데요, Success 를 매번 이렇게 감싸야하는게 많이 아쉽습니다.
ResponseBodyConverter 를 사용해서 data 만 꺼내서 반환하는 건 어떨까요 ?

[[Android] Retrofit2 Converter를 커스텀 해보기](https://antdevhistory.tistory.com/52)

Reply:
오 좋네요 저도 이 부분이 좀 아쉽다고 생각했는데 개선 방법이 떠오르지 않아 그냥 두었는데 참고 자료 보고 개선해보겠습니다!

Reply:
이 부분 작업하다가 생긴 질문이 있는데요!
원하시는 모습이 Success로 감싸지는 부분이 빠지고 특정 데이터 클래스가 ApiResult<>로 감싸지는 모양일까요?
```suggestion
interface ApiService {
    @GET("success")
    suspend fun getSuccess(): ApiResult<User>

    @GET("failure")
    suspend fun getFailure(): ApiResult<Room>
}
```

Reply:
넵 맞습니다 ~ 

`Success<User>` -> `User` 는 ResponseBodyConverter 가 수행할 거고,
`User` -> `ApiResult<User>` 는 CallAdapter 가 수행하겠네요